mockgcp/apply-proto-patches.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+set -x
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd ${REPO_ROOT}/mockgcp
+
+cd tools/patch-proto
+
+# Use our proto patch tool to add the missing maintenanceUpdatePolicy field for alloydb
+go run . --file ${REPO_ROOT}/mockgcp/third_party/googleapis/google/cloud/alloydb/v1beta/resources.proto --message Cluster <<EOF
```

Comment:
Mind sharing more context about this?

Reply:
Yes - good call.  Our public protos are missing the maintenanceUpdatePolicy field (I filed an internal bug for that).

The tool can patch the Protos.  That tool is sort of overkill (it uses treesitter), but it is something we're doing a lot of anyway because of CSV exporting for AI and when we're generating our types/mappers/etc.

What this lets us do is make progress when the public protos are not complete.  I think that's better than trying to maintain patches etc.  If we had more GitHub repos, we would probably just carry a fork of the public googleapis protos, but creating a repo is annoyingly difficult for us!

I'll add more comments etc.

Reply:
Thank you! Yeah, I saw the internal bug. May I ask how you find this missing field?